### PR TITLE
WIP - File naming convention changes

### PR DESCRIPTION
**Pull request by mairh**
- React components are in PascalCase now
- exported functions are in camelCase now
- Graphql file names are now same as the (mutation/query/subscription) function names.

- At the moment, I am getting a bunch of errors when running `yarn watch`. Not sure why it is complaining about `node_modules` folder. @vlasenko @mitjade Could you check on your end?

screen shot 2017-09-18 at 14 10 51

TODO: Refactor the functions in the `jsx`, `js` files to follow strict https://github.com/airbnb/javascript rules.

### CODE REVIEW

**Comment by mairh** [comment]
Ok. Found 1 prob. OSX doesn't do the lowercase / uppercase casting. I need to rename the file and then change it back to uppercase.

**Comment by vlasenko** [comment]
@mairh I see that you have way many files in lowercase instead of PascalCase, despite you reference them by PascalCase

**Comment by mairh** [comment]
This is still WIP. It is definitely possible that I have missed something as of now. Could you please point out in the code what are you referring to?

**Comment by mairh** [comment]
Yes, I see it now (for example `counter-jsx` vs `Counter.jsx`). I need to go through all the files that I have changed from lower case to PascalCase without the actual file name change

**Comment by vlasenko** [comment]
@mairh Yes

**Comment by mitjade** [comment]
@mairh Ware you able to figure out what is causing the errors?

**Comment by mitjade** [comment]
@mairh I see that the build passes. But I still get errors when I run this branch. Same as the ones you mentioned above.

**Comment by mairh** [comment]
yeah, I fixed the build errors related to filenames. However, I am a bit confused about the `node_module` errors I am getting in this branch.

I tried recreating the `yarn.lock` and deleted `node_modules` folder. Also, `package.json` file is exactly the same as in master.

@vlasenko Do you have any idea what could be the reason?

**Comment by mitjade** [comment]
@mairh I think I might be on to something. I started renaming file one by one and got this error when I mistyped `CounterView.jsx` and `CounterView.web.jsx`.

**Comment by mitjade** [comment]
`import CounterShow from '../components/CounterView';`

**Comment by mairh** [comment]
So `import CounterShow from '../components/CounterView';` works but not `import CounterView from '../components/CounterView';` ? Is that what you're trying to say?

**Comment by mitjade** [comment]
@mairh Still investigating.

**Comment by mitjade** [comment]
@mairh Found something!

**Comment by mairh** [comment]
Yeah. Apparently those .web extensions were added by default when you have npm auto import extension installed on visual studio code.

**Comment by mitjade** [comment]
@mairh Is it working for you now? I think the .jsx was the problem.

**Comment by mairh** [comment]
I didn’t check yet. I just saw the last commit on my phone. I will check it in 15 mins once I get hold on my laptop again.

**Comment by mairh** [comment]
@mitjade @vlasenko Code is working fine now. I will do a merge from the master and do the necessary updated in the tools folder.

**Comment by vlasenko** [comment]
@mairh Great, please go ahead

**Comment by mitjade** [comment]
@mairh Before merging merge with master after `flow` branch is merged, to fix some quote issues.

File: src/client/modules/pageNotFound/containers/PageNotFound.jsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import { Link } from 'react-router-dom';
 import Helmet from 'react-helmet';
 import { Button } from 'reactstrap';
-import PageLayout from '../../../app/page_layout';
+import PageLayout from '../../../app/PageLayout';
 
 const pageNotFound = () => (
```
Comment (mitjade) [inline src/client/modules/pageNotFound/containers/PageNotFound.jsx]: should we change this to `PageNotFound` to be consistent with imports of this component.

**Comment by mitjade** [comment]
@mairh Fixed all quote problems and template cli to correctly rename files.

File: src/client/modules/pageNotFound/containers/PageNotFound.jsx (hunk 1 above)
Comment (mairh) [reply src/client/modules/pageNotFound/containers/PageNotFound.jsx]: That is the Page wrapper component, not `PageNotFound` module

Comment (mitjade) [reply src/client/modules/pageNotFound/containers/PageNotFound.jsx]: `PageNotFound.jsx` is not a `PageNotFound` module?

Comment (mitjade) [reply src/client/modules/pageNotFound/containers/PageNotFound.jsx]: I am commenting line 7.

Comment (mairh) [reply src/client/modules/pageNotFound/containers/PageNotFound.jsx]: ah I see now. Yes, I will change that. I was paying attention to the line 5

Comment (mitjade) [reply src/client/modules/pageNotFound/containers/PageNotFound.jsx]: Sorry for the confusion.

**Comment by mairh** [comment]
@mitjade @vlasenko How do I solve the pre-commit prettier lint problem, after I merged the latest master into this PR (flow changes)

screen shot 2017-09-19 at 11 53 41

**Comment by mitjade** [comment]
@mairh I do not see latest `master` (with `flow`) being merged into this PR yet?

**Comment by mairh** [comment]
@mitjade Yes, It's because the pre-commit is failing due to the errors I reported and thus it doesn't allow you to push any changes.

**Comment by vlasenko** [comment]
@mairh @mitjade Let me handle that

**Comment by vlasenko** [comment]
@mairh @mitjade Merged master in and added `flow-typed` dir to `.eslintignore'

**Comment by vlasenko** [comment]
@mairh @mitjade If you are done, lets merge it into master

**Comment by mitjade** [comment]
@vlasenko @mairh I am all for merging this now.

**Comment by mairh** [comment]
Let's merge. We can continue further updates in a new PR

**Pull request merged**